src/components/MessageDisplay.svelte

```diff
@@ -41,9 +43,43 @@
       }
       items = items;
     });
+    const hideOrReplace = (i, bonkOrDeletion) => {
+      if ($ytcDeleteBehaviour === YtcDeleteBehaviour.HIDE) {
+        items[i].hidden = true;
+      } else if ($ytcDeleteBehaviour === YtcDeleteBehaviour.PLACEHOLDER) {
+        items[i].messageArray = bonkOrDeletion.replacedMessage;
+        items[i].deleted = true;
+      }
+    };
```

Comment:
You may need to do `items = items` at the end of the function to trigger svelte's reactivity for arrays.

Reply:
I tried `items = items`, `items[i] = items[i]` and a lot of other things and it wouldn't trigger reactivity until I made the `message.messageArray` to `messageArray` change

Reply:
Perhaps ```js
const before = items.slice(0, i);
const after = items.slice(i + 1);
items = [...before, {...items[i], hidden: true}, ...after];
```
will work?

I think svelte assumes immutable data when deciding reactivity.

Reply:
Ok, I tried both with a [demo project](https://gist.github.com/r2dev2/a01e185cd06e36efc574c2935c72ecdf), and seems like `[...before, newItem, ...after]` is the only one that works.

Reply:
This feels like way too many array operations lmao. Also, changing `hidden` already works as is, I'm assuming because it's a prop. That's why I made `messageArray` a prop as well.